app/controllers/dashboard_controller.rb

```diff
@@ -46,6 +47,10 @@ def current_courses
     current_user.courses.current_and_future
   end
 
+  def force_instructor_view?
+    params.key?(:teach) && user_signed_in?
```

Comment:
I don't think `user_signed_in?` is necessary, as a logged out user gets redirected to HomeController#index.

Reply:
Okay

Reply:
I've removed both user_signed_in and force_instructor_view, so I can now pass the teach parameter directly.